### PR TITLE
Docs for mendel config and mendel development

### PR DESCRIPTION
**Pull request by irae**
This PR is for documentation review, so we can land rebased and squashed without a lot of small revisions.

Any feedback is appreciated.

### CODE REVIEW

File: packages/mendel-config/Readme.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Mendel Config
+
+This is an internal package that helps Mendel normalize configuration defaults, on `.mendelrc` or on `package.json`. It is used by many Mendel packages to make sure to make sure all packages use the same defaults and same merging logic.
```
Comment (tufandevrim) [inline packages/mendel-config/Readme.md]: `to make sure`

File: packages/mendel-config/Readme.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+#### 1. Merging by precedence:
+
+Configuration precedence is, from order to strongest to weakest:
+
+  1. Passing configuration as an JavaScript object
+  2. Configuration on `.mendelrc` or `package.json`
+  3. Mendel defaults
+
+There is only exception is `basedir`. `basedir` has different meanings in different meaning depending on where you declare it:
```
Comment (tufandevrim) [inline packages/mendel-config/Readme.md]: `has different meanings depending on where you declare it`

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by irae** [comment]
Thanks for the review

**Pull request merged**